sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -244,11 +258,31 @@ statement
     | SET TIME ZONE interval                                           #setTimeZone
     | SET TIME ZONE timezone=(STRING | LOCAL)                          #setTimeZone
     | SET TIME ZONE .*?                                                #setTimeZone
+    | SET configKey (EQ configValue)?                                  #setConfiguration
     | SET .*?                                                          #setConfiguration
+    | RESET configKey?                                                 #resetConfiguration
     | RESET .*?                                                        #resetConfiguration
     | unsupportedHiveNativeCommands .*?                                #failNativeCommand
     ;
 
+configKey
+    : configIdentifier (('.' | ':') configIdentifier
+        // The two semantic predicates make sure that no token on the hidden channel
+        // (e.g., spaces) exists between the config separator ('.' or ':').
+        {!isHidden(-3)}? {!isHidden(-2)}?)*
+    | quotedIdentifier
+    ;
+
+configValue
+    : MINUS? vaule=.*?
```

Comment:
hmm, why `.*` can't match MINUS?

Reply:
Yea, this is one of the last few things I need to check...I'm still looking into why. Please let me know if you know something about this behaivour.

Reply:
```
    | SET configKey (EQ value=.*)?                                     #setConfiguration
```
This will lead to a failure below;
```
[info] - Checks if SET/RESET can accept all the configurations *** FAILED *** (6 seconds, 11 milliseconds)
[info]   == FAIL: Plans do not match ===
[info]   !SetCommand (spark.sql.avro.deflate.level,Some(1))   SetCommand (spark.sql.avro.deflate.level,Some(-1)) (PlanTest.scala:157)
[info]   org.scalatest.exceptions.TestFailedException:
```

But, if we change it like this, it works well...;
```
    | RESET configKey?                                                 #resetConfiguration

configValue
    : vaule=.*?
    ;
```